Feat/KFS-1598 KFS-1599 escaped use statement

Guttz opened the pull request:
Release Notes
-------------
<!--
If this PR introduces any user-facing changes, please document them below. Please delete any unused section titles and placeholders.
Please match the style of previous release notes: https://docs.confluent.io/confluent-cli/current/release-notes.html
-->
New Features
- Support escaped characters for catalogs and databases set with "USE" in `confluent flink shell`


Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
<!--
Briefly describe **what** you have changed and **why**.
Optionally include your implementation strategy.
-->
As Specification for SQL Clients: USE statements in `confluent flink shell` now support escaped catalog and database using backticks.


References
----------
<!--
Copy and paste links to tickets, related PRs, GitHub issues, etc.
-->

Test & Review
-------------
<!--
Has it been tested? How?
Copy and paste any instructions or steps that can save the reviewer time.
-->

brianstrauch commented:
> USE statements in confluent flink shell now support escaped catalog and database using backticks

How about:
```
Support escaped characters for catalogs and databases selected with the "USE" statement in `confluent flink shell`.
```

brianstrauch commented on pkg/flink/internal/store/store_test.go:
```diff
@@ -522,47 +522,153 @@ func (s *StoreTestSuite) TestParseSetStatementError() {
-	key, value, _ := parseUseStatement("USE CATALOG c;")
-	assert.Equal(s.T(), flinkconfig.KeyCatalog, key)
-	assert.Equal(s.T(), "c", value)
+	catalog, database, _ := parseUseStatement("USE CATALOG c;")
```
I like the original API better. We should return a type and name `("CATALOG", "c")` to make things easier if we decide to add more use-able types in the future

Guttz replied:
Me too... However, use can now return both catalog and database, and it couldn't in the previous version. That means, I had to either:

1 - Return two strings
2 - Return an array with two positions
3 - Return an object

Since SQL is really stable and very rarely changes, this function will probably return two values for a long time. My thought was that 2 and 3 would require too much boil code for a function that only returns two values. I'd be ok with all 3 but that's why I personally prefer 1.

brianstrauch replied:
> use can now return both catalog and database

Ah, didn't realize that :+1:

Guttz commented:
> > USE statements in confluent flink shell now support escaped catalog and database using backticks
> 
> How about:
> 
> ```
> Support escaped characters for catalogs and databases selected with the "USE" statement in `confluent flink shell`.
> ```

Updated it and just changed select to set :)
